unified_planning/engines/oversubscription_planner.py

```diff
@@ -124,7 +134,10 @@ def _solve(
             new_problem = problem.clone()
             new_problem.clear_quality_metrics()
             for g in t[1]:
-                new_problem.add_goal(g)
+                if isinstance(g, tuple) and list(map(type, g)) == [TimeInterval, FNode]:
+                    new_problem.add_timed_goal(g[0], g[1])
```

Comment:
```suggestion
                if isinstance(g, tuple):
                    assert list(map(type, g)) == [TimeInterval, FNode]
                    new_problem.add_timed_goal(g[0], g[1])
```
Here i would put the second part of the `and` as an `assert`, otherwise, if the second part of the and fails (but the first one doesn't), it gets added as a goal and the error report is "tuple does not have environment field" in the `add_goal` method, which is quite weird.